src/polyswarmd/views/event_message.py

```diff
@@ -137,12 +137,13 @@ def channel_events(ws, guid):
         for evt in filter_events:
             filter_manager.register(offer_msig.eventFilter, evt)
 
-        with filter_manager.fetch() as results:
-            for messages in results:
+        def callback(messages):
+            for msg in messages:
                 if ws.closed:
                     raise RuntimeError("WebSocket is closed")
-                for msg in messages:
-                    return ws.send(msg)
+                return ws.send(msg)
+
+        filter_manager.pipe_events(callback)
```

Comment:
Nice!

Just one question: Is this `pipe_events()` blocking?
I mean: If not, will work up to when `filter_manager` got garbage-collected, as this function end will release the pointer.

Reply:
If there are any (non-weak) references to an object the garbage collector shouldn't collect the object.

The code which previously used `weakref` in FilterManager.py was intended to prevent reference _cycles_ from occurring inside the gevent ev, but we removed it because we stopped keeping per-entry statistics and a message handler (which had references to one another). 

Today, I don't believe there are objects with reference cycles, but if you find any definitely post the `gc` output.

Reply:
So `filter_manager` will be garbage-collected on this function end. After that, the callback function will not be called anymore.

I think that this is a problem. Is there something that I am missing here?